core/src/main/scala/cats/data/NonEmptyVector.scala

```diff
@@ -38,27 +42,22 @@ final case class NonEmptyVector[A] private (toVector: Vector[A]) {
   def filter(f: A => Boolean): Vector[A] = toVector.filter(f)
 
   /**
-   * Append another NonEmptyVector to this
+   * Alias for [[concat]]
    */
-  def concat(other: NonEmptyVector[A]): NonEmptyVector[A] = NonEmptyVector(toVector ++ other.toVector)
-
-  /**
-   * Alias for concat
-   */
-  def ++(other: NonEmptyVector[A]): NonEmptyVector[A] = concat(other)
+  def ++(other: Vector[A]): NonEmptyVector[A] = concat(other)
 
   /**
-   * Append another Vector to this
+   * Append another `Vector` to this, producing a new `NonEmptyVector`.
    */
-  def concat(other: Vector[A]): NonEmptyVector[A] = NonEmptyVector(toVector ++ other)
+  def concat(other: Vector[A]): NonEmptyVector[A] = new NonEmptyVector(toVector ++ other)
 
   /**
-   * Alias for concat
+   * Append another `NonEmptyVector` to this, producing a new `NonEmptyVector`.
    */
-  def ++(other: Vector[A]): NonEmptyVector[A] = concat(other)
+  def concatNEV(other: NonEmptyVector[A]): NonEmptyVector[A] = new NonEmptyVector(toVector ++ other.toVector)
```

Comment:
No reason I can think of. I wouldn't have a huge preference either way because they are functionally equivalent.